contracts/protocol/facets/DisputeResolverHandlerFacet.sol

```diff
@@ -188,40 +193,184 @@ contract DisputeResolverHandlerFacet is IBosonAccountEvents, ProtocolBase {
+            preUpdateDisputeResolverCheck(_disputeResolver.id, _disputeResolver.clerk, lookups);
+            require(_disputeResolver.clerk != address(0), INVALID_ADDRESS);
+
+            // If clerk address exists, clerk address owner must approve the update to prevent front-running
+            disputeResolverPendingUpdate.clerk = _disputeResolver.clerk;
+            needsApproval = true;
+        }
+
+        if (needsApproval) {
+            // Notify watchers of state change
+            emit DisputeResolverUpdatePending(_disputeResolver.id, disputeResolverPendingUpdate, sender);
+        }
+
+        bool updateApplied;
+
+        if (_disputeResolver.treasury != disputeResolver.treasury) {
+            require(_disputeResolver.treasury != address(0), INVALID_ADDRESS);
+
+            // Update treasury
+            disputeResolver.treasury = _disputeResolver.treasury;
+
+            updateApplied = true;
+        }
+
+        if (_disputeResolver.escalationResponsePeriod != disputeResolver.escalationResponsePeriod) {
+            // Escalation period must be greater than zero and less than or equal to the max allowed
+            require(
+                _disputeResolver.escalationResponsePeriod > 0 &&
+                    _disputeResolver.escalationResponsePeriod <= protocolLimits().maxEscalationResponsePeriod,
+                INVALID_ESCALATION_PERIOD
+            );
+
+            // Update escalation response period
+            disputeResolver.escalationResponsePeriod = _disputeResolver.escalationResponsePeriod;
+
+            updateApplied = true;
+        }
+
+        if (keccak256(bytes(_disputeResolver.metadataUri)) != keccak256(bytes(disputeResolver.metadataUri))) {
+            // Update metadata URI
+            disputeResolver.metadataUri = _disputeResolver.metadataUri;
+
+            updateApplied = true;
+        }
+
+        if (updateApplied) {
+            // Notify watchers of state change
+            emit DisputeResolverUpdateApplied(
+                _disputeResolver.id,
+                disputeResolver,
+                disputeResolverPendingUpdate,
+                sender
+            );
+        }
+    }
+
+    /**
+     * @notice Opt-in to a pending dispute resolver update
+     *
+     * Emits a DisputeResolverUpdateApplied event if successful.
+     *
+     * Reverts if:
+     * - The dispute resolver region of protocol is paused
+     * - Addresses are not unique to this dispute resolver
+     * - Caller address is not pending update for the field being updated
+     * - No pending update exists for this dispute resolver
+     *
+     * @param _disputeResolverId - disputeResolver id
+     * @param _fieldsToUpdate - fields to update, see DisputeResolverUpdateFields enum
+     */
+    function optInToDisputeResolverUpdate(
+        uint256 _disputeResolverId,
+        DisputeResolverUpdateFields[] calldata _fieldsToUpdate
+    ) external disputeResolversNotPaused nonReentrant {
```

Comment:
We aren't making any calls to external, untrusted contracts here. Do we need `nonReentrant`?

Reply:
Ok. Given that a scenario exists, however ridiculous, I agree we should leave it in. 

This is an interesting potential reentrancy -- not reentering the same method, but calling back in to a different method in order to leverage the current msgSender() -- that almost makes every method potentially needing `nonReentrant` on it.